### ISSUE TITLE
EVP_CIPHER_CTX_rand_key() does not validate that ctx->cipher is set

### ISSUE THREAD

**Issue by Hlipa** [openssl/openssl#19793]
ctx->cipher is accessed directly without check on line 1350 which is common in other functions:

```
1348  int EVP_CIPHER_CTX_rand_key(EVP_CIPHER_CTX *ctx, unsigned char *key)
1349  {
1350      if (ctx->cipher->flags & EVP_CIPH_RAND_KEY)
1351          return EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_RAND_KEY, 0, key);

```
See for example:
```
1368  int EVP_CIPHER_CTX_copy(EVP_CIPHER_CTX *out, const EVP_CIPHER_CTX *in)
1369  {
1370      if ((in == NULL) || (in->cipher == NULL)) {
```
Note that NULL checks are not consistent in evp_enc.c and  ERR_raise() is missing on any places too.

**Comment by t8m** [comment]
In general it is not safe to assume with the OpenSSL API calls that calling functions with obviously required parameters set to NULL will return error and not crash. Yes, currently we prefer adding these NULL checks, but it is not a bug if a previously existing function does not have such check.

The exception are the _free() functions where a missing NULL check would be a bug.

Even more so this is true about missing ERR_raise() calls in case of caller misusing the API or other errors. In many cases adding ERR_raise() calls can be even breaking existing callers that do not expect errors to be risen.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
